### DISCUSSION TITLE
Booleans annotation

### DISCUSSION THREAD

**Discussion by alexxRizz** [junit-pioneer/junit-pioneer#717] [Ideas]
Hi! I feel strange there is no such annotation in this cool library. Something like this:
```kotlin
@Target(AnnotationTarget.VALUE_PARAMETER)
@Retention(AnnotationRetention.RUNTIME)
@CartesianArgumentsSource(BooleanArgumentsProvider::class)
annotation class Booleans

class BooleanArgumentsProvider : CartesianParameterArgumentsProvider<Boolean> {

  override fun provideArguments(context: ExtensionContext, parameter: Parameter): Stream<Boolean> =
    booleanArrayOf(true, false).asSequence().asStream()

}
```
Usage:
```kotlin
@CartesianTest
fun test(@Booleans flag1: Boolean, @Booleans flag2: Boolean) {
}
```

**Comment by beatngu13** [comment]
Hi @alexxRizz,

you mean as a shorthand for:

```java
@CartesianTest
void test(
		@Values(booleans = {true, false}) boolean x,
		@Values(booleans = {true, false}) boolean y) {
	System.out.println(x + " and " + y + " = " + (x && y));
}
```

And/or more generally a `@BooleanSource` with a corresponding [`ArgumentsProvider`](https://junit.org/junit5/docs/current/api/org.junit.jupiter.params/org/junit/jupiter/params/provider/ArgumentsProvider.html)?

Related: https://github.com/junit-team/junit5/issues/2878

**Reply by alexxRizz** [reply]
In `@CartesianTest` shorthand for all possible values of boolean, as already existing `@Enum` annotation, which provides all possible values for enum.

**Comment by alexxRizz** [comment]
For example, in C# similar approach exists in NUnit:
```c#
[Test]
public void MyBoolTest([Values] bool value1, [Values] bool value2)
{
}
```
